src/main/java/hudson/plugins/jacoco/JacocoBuildAction.java

```diff
@@ -359,4 +355,14 @@ public final PrintStream getLogger() {
 	    // does not run the construct and thus leaves the transient variables empty
 	    return System.out;
 	}
+
+	@Override
+	public void onAttached(Run<?, ?> run) {
+		this.owner = run;
```

Comment:
Note that this means you could delete the `owner` parameter from the constructor (assuming it is not used within the dynamic scope of the constructor).